lib/OpenQA/Shared/Controller/Auth.pm

```diff
@@ -25,7 +25,10 @@ sub check {
 
     my $req = $self->req;
     if ($self->app->config->{no_localhost_auth}) {
-        return 1 if $self->tx->remote_address eq '127.0.0.1';
+
+        # IPv4 and IPv6 should be treated the same
+        my $address = $self->tx->remote_address;
+        return 1 if $address eq '127.0.0.1' || $address eq '::1';
```

Comment:
This feels a little awkward (shouldn't there be an API for this?) but makes sense.

Reply:
Can we check this in a unit test by any chance?

Reply:
There is no API, and a test would be rather tricky. At least i can't think of a good (and reliable) way to test it at the moment.

Reply:
An API is being discussed upstream now though, so maybe we can improve this in the future.